iphone/cli/commands/_build.js

```diff
@@ -3825,6 +3825,24 @@ iOSBuilder.prototype.createXcodeProject = function createXcodeProject(next) {
 			this.logger.debug(__('Writing %s', dest.cyan));
 			fs.ensureDirSync(parent);
 			fs.writeFileSync(dest, contents);
+
+			// Xcode 11 requires the workspacedata file to exist or else it doesn't properly
+			// rebuild the app if the Xcode project changes
+			const xcworkspaceDir = path.join(this.buildDir, this.tiapp.name + '.xcodeproj', 'project.xcworkspace');
```

Comment:
It should also be added in the iOS Xcode template, so users don't need to create this file on every clean build.

Reply:
Also, I think it's somehow causing a force-rebuild. At least, all source SDK files (*.m) are recompiled.

Reply:
And finally, the logs don't work again after running it 2-3 more times.

Reply:
Good point, I can add the xcworkspace to the Titanium.xcodeproj dir.

You are correct, the build is doing a rebuild because the Xcode project keeps changing because the log server port is defined in as a GCC define.

Gah, so the fix doesn't work. I'll have to take a peek.

Without rehashing all of the reasons that got us into this place, we have a couple options:

1. Have the iOS build attempt to connect to the app and verify the app id, before bothering to pick a random port. The build used to do this, but it only works if the app is foregrounded.

2. Move the log server port from a GCC define to a json file that is parsed at runtime. We would probably still need to run xcodebuild, but it should be super fast since no code changed.

3. Rip out the log server and have NSLog() write to a log file for simulator builds only, then the iOS build would tail that file. I believe this is how it used to work. Device builds would continue to use the log server.

4. Something else?

I personally like option 3. Simple to implement.

Reply:
If device builds work fine now, 3. would be a good option. But wouldn't the file inside the `.xcodeproj` fix it already? And for the forced change, either the file can be unmarked (like some others already).

Reply:
From what I can tell, the `project.xcworkspace` is requried by Xcode 11 for xcodebuild to detect changes to project build settings like the gcc defs. So, we need this file regardless.

When it builds your app and launches the app in the simulator, the build will have already selected the log server port. On a subsequent build, the port is in use, thus a new random port is selected and set in the gcc def. Now xcodebuild detects the setting change and builds only the affected files which unfortunately is basically all of them and that's the rebuild you are seeing.

If you simply kill the app in the simulator (not backgrounding the app or killing the sim, but kill the app), then when you rebuild, the build will use the same log server port and no rebuild is necessary.

Yeah, the `project.xcworkspace` should maybe be managed by the iOS build and unmarked for differential builds, but given this file never changes, I sorta cheated and just make sure it exists.